### PR TITLE
cloudinit to customspec support

### PR DESCRIPTION
**Pull request by karmab**
this PR adds a function to parse a user_data cloudinit like string ( as could be sent from foreman) and convert it to an hash usable as customisationspec

### CODE REVIEW

**Comment by karmab** [comment]
did you get a chance to review and merge?

File: lib/fog/vsphere/requests/compute/vm_clone.rb [hunk 1]
```diff
@@ -165,7 +165,7 @@ def vm_clone(options = {})
             cust_ip_settings ||= RbVmomi::VIM::CustomizationIPSettings.new("ip" => RbVmomi::VIM::CustomizationDhcpIpGenerator.new())
             cust_ip_settings.dnsDomain = cust_domain
             cust_global_ip_settings = RbVmomi::VIM::CustomizationGlobalIPSettings.new
-            cust_global_ip_settings.dnsServerList = cust_ip_settings.dnsServerList
+	    cust_global_ip_settings.dnsServerList = cust_ip_settings.dnsServerList
```
Comment (nirvdrum) [inline lib/fog/vsphere/requests/compute/vm_clone.rb]: Small issue, but please don't replace spaces with tabs.  The code no longer lines up depending on tab-spacing.  If you can fix and rebase, that'd be helpful.

**Comment by karmab** [comment]
did you get a chance to review and merge?

**Comment by nirvdrum** [comment]
I wasn't notified that you had had addressed my comment, so no.  It'd be much politer to inform me of that than to follow up with the same exact question.  Now that I'm aware, I'll plan on reviewing again.

**Comment by geemus** [comment]
Yeah, I really wish github would ping you when commits were added to stuff you were discussing. I've asked them about it, so :fingers_crossed:

File: lib/fog/vsphere/requests/compute/cloudinit_to_customspec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          custom_spec['dnsSuffixList']               =  user_data['domain'] if user_data.key?('domain')
+          custom_spec['time_zone']                   =  user_data['timezone'] if user_data.key?('timezone')
+          custom_spec           
+        end
+      end
+
+      class Mock
+        def cloudinit_to_customspec(user_data)
+          raise ArgumentError, "user_data cant be nil" if user_data.nil?
+          true
```
Comment (nirvdrum) [inline lib/fog/vsphere/requests/compute/cloudinit_to_customspec.rb]: We try to make the mocks reflect the real operations as closely as possible.  Granted, the codebase is pretty scattershot on this, but it is the goal we aspire to.  Is there a reason you couldn't return the same hash as the real implementation?  It doesn't look like you're making a 3rd party API call, so I'd just even delegate the mock to the real in this case.

**Comment by nirvdrum** [comment]
@karmab This looks good to go.  Thanks for being patient.  I have one question/comment about the mocks.  If it's not something you can get to, I'll merge and fix it up myself.  If you can, great.  But I'd like you to have an opportunity to review it to be sure what I'm suggesting or asking makes sense.

**Comment by nirvdrum** [comment]
I fixed the merge conflict and supplied a mock implementation.  This has now been merged. Thanks!

**Comment by geemus** [comment]
Thanks!

**Pull request merged**